Add provisioner for base node pool

AlmightyYakob opened the pull request:
This adds the karpenter provisioner required for the `base` node pool.

It doesn't seem there's any code here for the autoscaling group associated with the base node pool, so I guess that will just need to be manually deprovisioned?

AlmightyYakob commented:
@zackgalbreath a few questions:

* I currently only see `t3.small` used in the base node group. Do you think it's fine to just keep this, or should we expand it?
* I restricted to only use on-demand, as I don't think we'll want to deal with the logistics of spot instances in the base node pool. However, if we are okay with that, we could likely save a decent amount of cost.
* I didn't enforce any limits yet on this, but from looking at the ASG in EC2, it seems like 64 is the max. Should I apply a resource limit of 64*`t3.small`?

zackgalbreath commented:
> * I currently only see `t3.small` used in the base node group. Do you think it's fine to just keep this, or should we expand it?
> * I restricted to only use on-demand, as I don't think we'll want to deal with the logistics of spot instances in the base node pool. However, if we are okay with that, we could likely save a decent amount of cost.
> * I didn't enforce any limits yet on this, but from looking at the ASG in EC2, it seems like 64 is the max. Should I apply a resource limit of 64*`t3.small`?

In order to reduce the number of variables, let's keep it the same as what we're currently doing with Cluster Autoscaler for the time being: up to 64 t3.small on-demand instances.

AlmightyYakob commented:
I think the migration plan here should be:

1. Merge this PR
2. Apply the provisioner
3. gradually scale down the max number of nodes in the base autoscaling group, making sure karpenter is picking up the pods and provisioning correctly
4. Delete the autoscaling group

AlmightyYakob commented:
> I think the migration plan here should be:
> 
> 1. Merge this PR
> 2. Apply the provisioner
> 3. Gradually scale down the max number of nodes in the base autoscaling group, making sure karpenter is picking up the pods and provisioning correctly
> 4. Delete the autoscaling group

@zackgalbreath How does this sound to you? If it sounds good, can I get an approval? (already addressed your above comment).

zackgalbreath commented:
> > I think the migration plan here should be:
> > 
> > 1. Merge this PR
> > 2. Apply the provisioner
> > 3. Gradually scale down the max number of nodes in the base autoscaling group, making sure karpenter is picking up the pods and provisioning correctly
> > 4. Delete the autoscaling group
> 
> @zackgalbreath How does this sound to you? If it sounds good, can I get an approval? (already addressed your above comment).

Sounds good to me!
